serde/src/de/impls.rs

```diff
@@ -295,6 +297,33 @@ impl Deserialize for String {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+#[cfg(feature = "std")]
+impl Deserialize for Box<CStr> {
```

Comment:
Please remove this impl unless you can implement it safely. The internal representation of CStr is not stable and there are plans to change it in the future.